opm/simulators/wells/BlackoilWellModelGeneric.cpp

```diff
@@ -2374,4 +2374,20 @@ runWellPIScaling(const int timeStepIdx,
     this->last_run_wellpi_ = timeStepIdx;
 }
 
+
+bool
+BlackoilWellModelGeneric::
+guideRateUpdateIsNeeded() const {
+    int update = 0;
+    for (const auto& well : well_container_generic_) {
+        if (well->changedToOpenThisStep()) {
+            update = 1;
+            break;
+        }
+    }
+    update = comm_.max(update);
+    return update;
```

Comment:
I might possibly consider writing this in terms of [`std::any_of()`](https://en.cppreference.com/w/cpp/algorithm/all_any_none_of)
```C++
const auto need_update =
    std::any_of(this->well_container_generic_.begin(),
                this->well_container_generic_.end(),
    [](const WellInterfaceGeneric* well)
{
    return well->changedToOpenThisStep();
});

return this->comm_.max(static_cast<int>(need_update));
```
This, I think, is more expressive than a raw loop with a nested conditional.